govcd/nsxt_edgegateway_bgp_configuration_test.go

```diff
@@ -0,0 +1,70 @@
+//go:build network || nsxt || functional || openapi || ALL
+// +build network nsxt functional openapi ALL
+
+package govcd
+
+import (
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	. "gopkg.in/check.v1"
+)
+
+func (vcd *TestVCD) Test_NsxEdgeBgpConfiguration(check *C) {
+	skipNoNsxtConfiguration(vcd, check)
+	skipOpenApiEndpointTest(vcd, check, types.OpenApiPathVersion1_0_0+types.OpenApiEndpointEdgeBgpConfig)
+
+	org, err := vcd.client.GetOrgByName(vcd.config.VCD.Org)
+	check.Assert(err, IsNil)
+	nsxtVdc, err := org.GetVDCByName(vcd.config.VCD.Nsxt.Vdc, false)
+	check.Assert(err, IsNil)
+	edge, err := nsxtVdc.GetNsxtEdgeGatewayByName(vcd.config.VCD.Nsxt.EdgeGateway)
+	check.Assert(err, IsNil)
+
+	// Switch Edge Gateway to use dedicated uplink for the time of this test and then turn it off
+	err = switchEdgeGatewayDedication(edge, true) // Turn on Dedicated Tier 0 gateway
+	check.Assert(err, IsNil)
+	defer switchEdgeGatewayDedication(edge, false) // Turn off Dedicated Tier 0 gateway
+
+	// Get and store existing BGP configuration
+	bgpConfig, err := edge.GetBgpConfiguration()
+	check.Assert(err, IsNil)
+	check.Assert(bgpConfig, NotNil)
+
+	// Disable BGP
+	err = edge.DisableBgpConfiguration()
+	check.Assert(err, IsNil)
+
```

Comment:
assert that disable happen missing

Reply:
I saw same at the end of test, so not sure why this needed here

Reply:
This first `disable` happened just to be sure that it is disabled before testing configuration. BGP gets enabled by default in some cases.